### PR TITLE
TODO's and FIXME's [PART 1]

### PR DESCRIPTION
**Pull request by vrothberg**
Fix a number of TODO's and FIXME's across the code base.  Please refer to the individual commits for details.

```release-note
* Fix an issue in `podman image mount` where the pretty table for multiple images was not printed.
* Add auto-completion to `podman search --filter`.
```

### CODE REVIEW

File: cmd/podman/common/completion.go [hunk 1]
```diff
@@ -496,6 +497,11 @@ func AutocompleteImages(cmd *cobra.Command, args []string, toComplete string) ([
 	return getImages(cmd, toComplete)
 }
 
+// AutocompleteImageSearchFilters - Autocomplate `search --filter`.
+func AutocompleteImageSearchFilters(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return libimageDefine.SearchFilters, cobra.ShellCompDirectiveDefault
```
Comment (Luap99) [inline cmd/podman/common/completion.go:502]: should use ShellCompDirectiveNoFileComp since you cannot use paths for this option

**Comment by vrothberg** [comment]
Restarted the flake - the flake has been fixed in the meantime.

**Pull request merged**